api-consumption/src/main/java/org/zalando/nakadi/service/subscription/state/ClosingState.java

```diff
@@ -159,7 +159,7 @@ private void freePartitions(final Collection<EventTypePartition> keys) {
                 }
             }
         }
-        getZk().runLocked(() -> getZk().transfer(getSessionId(), keys));
+        getZk().transfer(getSessionId(), keys);
```

Comment:
@adyach Either we need to have feature toggle to use/not use `runLocked` call here, or (as a first iteration) leave `runLocked` call, as it won't be possible to seamlessly migrate to lockless approach. The same goes to SteamingState call.
I propose to use featureToggles